admin_manual/enterprise/external_storage/s3_swift_as_primary_object_store_configuration.rst

```diff
@@ -1,159 +1,167 @@
+
+`The S3 Object Storage app`_ (``files_primary_s3``) is the preferred way to provide S3 storage support,
+However, `the Object Storage Support app`_ (``objectstore``) is still available.
+
+.. note:: OpenStack Swift has been deprecated.
+
+When using ``files_primary_s3``, the Amazon S3 bucket needs to be created manually according to the developer documentation, and versioning needs to be enabled.
+
+.. note:: ownCloud GmbH provides consulting for migrations from ``objectstore`` to ``files_primary_s3``.
+
+.. note:: The source of the S3 Object Storage app `is available on gitHub`_.
 
 Implications
 ------------
-This configuration needs to be applied before the first login of any user – including the 
-admin user. Otherwise, you will run into an error condition, because ownCloud cannot find 
-the user's files anymore.
 
-ownCloud in object store mode expects exclusive access to the object store 
-container, because it only stores the binary data for each file. The metadata 
-are kept in the local database for performance reasons.
+- Apply this configuration before the first login of any user – including the admin user; otherwise, ownCloud can no longer find the user's files.
+
+- ownCloud, in "object store" mode, expects exclusive access to the object store container, because it only stores the binary data for each file. While in this mode, ownCloud stores the metadata in the local database for performance reasons.
 
-The current implementation is incompatible with any app that uses 
-direct file I/O and circumvents the ownCloud virtual filesystem. That includes 
-Encryption and Gallery. Gallery stores thumbnails directly in the 
-filesystem, and Encryption causes severe overhead because key files need 
-to be fetched in addition to any requested file.
+- The current implementation is incompatible with any app that uses direct file I/O (input/output) as it circumvents the ownCloud virtual filesystem. Two excellent examples are:
+
+  #. **The Encryption app:** It fetches critical files in addition to any requested file, which results in significant overhead.
+  #. **The Gallery app:** It stores thumbnails directly in the filesystem.
 
 Configuration
 -------------
 
-Look in ``config.sample.php`` for a example configurations. Copy the 
-relevant part to your ``config.php`` file. Any object store needs to implement
-``\\OCP\\Files\\ObjectStore\\IObjectStore`` and can be passed parameters in the
-constructor with the ``arguments`` key:
+Look in ``config.sample.php`` for example configurations.
+Copy the relevant part to your ``config.php`` file.
+Any object store needs to implement ``\\OCP\\Files\\ObjectStore\\IObjectStore``, and can be passed parameters in the constructor with the ``arguments`` key, as in the following example:
 
-::
+.. code-block:: php
 
-    'objectstore' => [
-        'class' => 'Implementation\\Of\\OCP\\Files\\ObjectStore\\IObjectStore',
-        'arguments' => [
-            ...
+    $CONFIG = [
+        'objectstore' => [
```

Comment:
I hope that this is not a stupid question, but...
Why is the key ``objectstore`` embedded in an $CONFIG array?
If I look to key ``apps_paths``, there is also no embedding $CONFIG array
Is this really the case here - looks so strange

Reply:
@patrickjahns mind to give an insight here?

Reply:
I wrapped it as the majority of examples did. I could well have been wrong.

Reply:
Do you have an example that I can refer to, @mmattel?

Reply:
sure, just check the key ``apps_paths`` in config.sample.php.
It has no wrapping key around the actualy one.